### PR TITLE
Fix Doc spelling and broken links, removed warnings about using main

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Added explanation on how to add custom ONNX or Krnl Op, including the make file command to regenerate the local doc. Saw that this info was missing from our current docs

Signed-off-by: Alexandre Eichenberger <alexe@us.ibm.com>

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
Also removed the temporary warnings about now using main instead of master. I believe most of our users are good now with the new main.

File: docs/HowToAddAnOperation.md [hunk 1]
```diff
@@ -64,4 +64,17 @@ enables a new test, `test_erf_cpu`, where all inputs may accommodate a fully dyn
...
 
+# How to add a custom ONNX or Krnl operation
+
+Sometime, we need to add support for a custom ONNX Operation (for example to support older version of a current ONNX operation) or for a custom Krnl Operation. 
+
+Custom ONNX Operations are added in the `src/Dialects/ONNX/ONNXOps.td`. Once added there, this new operation will augment all of the spec-defined ONNX operations. Support is needed for this op and is added following the same steps as above.
+
+Custom Krnl Operations are added in the `src/Dialects/KrnlOps.td` and support is also needed to implement the new operation. Handling of the operation itself (e.g. custom builders) are handled in the KrnlOps.cpp in the same folder. Eventually, this operation will need to be converted to standard MLIR operation. Depending on whether this operation is lowered to affine/standard dialects or LLVM dialects, such code will be required, respectively, in the `src/Conversion/KrnlToAffine` or `src/Conversion/KrnlToLLVM` subdirectories.
```
Comment (etiotto) [inline docs/HowToAddAnOperation.md:73]: typo:  should be `src/Dialects/Krnl/KrnlOps.td`

**Review by etiotto — APPROVED** [review]
LGTM 
I just have one comment about a typo.

File: docs/HowToAddAnOperation.md (hunk 1 above)
Comment (AlexandreEichenberger) [reply docs/HowToAddAnOperation.md:73]: good point, let me update

**Pull request merged**